### PR TITLE
[v2] Introduce frontend:dev:watcher command which is non blocking

### LINKED ISSUE #1169: unexpected behavior for "frontend:dev" config option, does not run concurrently
#####################################################
     **V1 users: If you have a technical issue, please do not open a bug this way!** 
                      Please use the `wails issue` command!
          If you do not do this then the issue may be closed automatically.                      

 NOTE: If your bug is related to Windows, make sure you read 
 the [Windows Developer Guide](https://wails.app/guides/windows/)
#####################################################

**Description**
The documentation for this field is as follows:

```
"[This command is run in a separate process on `wails dev`. Useful for 3rd party watchers]",
```

To me, this sets up the expectation that this command will run _concurrently_ with `wails dev`. Especially considering the example given is "3rd party watchers".  most watcher scripts will launch and never exit, watching for changes.

I'd be happy to give a PR for this change, if someone can give me a jumping off point in the source code. thank you!

### PR DESCRIPTION
**Pull request by stffabi**
Closes #1169

### CODE REVIEW

**Comment by stffabi** [comment]
This PR assumes that `frontend:dev` is really only for watchers and not the build equivalent in dev mode.

**Comment by misitebao** [comment]
@stffabi 
I tested your code and can solve the problem mentioned in (#1169), but I accidentally discovered other problems (not related to your code) during the test😂. I will open another issue to explain this problem.

**Comment by stffabi** [comment]
Great thanks for testing @misitebao. So I think we are good to go with this one, if @leaanthony is fine with the proposal.

**Comment by misitebao** [comment]
@stffabi 
Here are the new issues found: https://github.com/wailsapp/wails/issues/1175

**Comment by quinn** [comment]
@stffabi I was able to test, works as expected! thank you 🙏

**Comment by leaanthony** [comment]
> @stffabi Here are the new issues found: #1175

I think this is a slightly different issue and I will comment on that ticket :-)

File: website/docs/reference/project-config.mdx [hunk 1]
```diff
@@ -13,7 +13,8 @@ The project config resides in the `wails.json` file in the project directory. Th
 	"reloaddirs": "[Additional directories to trigger reloads (comma separated), this is only used for some advanced asset configurations]",
 	"frontend:install": "[The command to install node dependencies, run in the frontend directory - often `npm install`]",
 	"frontend:build": "[The command to build the assets, run in the frontend directory - often `npm run build`]",
-	"frontend:dev": "[This command is the dev equivalent of frontend:build]",
+	"frontend:dev": "[This command is the dev equivalent of frontend:build. If not specified falls back to frontend:build]",
```
Comment (leaanthony) [inline website/docs/reference/project-config.mdx:16]: Does it really fallback to `frontend:build` ?

**Pull request merged**